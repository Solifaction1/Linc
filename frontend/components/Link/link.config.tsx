import React from 'react'

type propsConfig = {
	href: string
	linkType?: 'navigation' | 'external' | string
	size?: 'sm' | 'md' | 'lg' | string
	disabled?: boolean
	className?: string
	handleClick?: React.MouseEventHandler<HTMLButtonElement>
	children?: React.ReactNode
	label?: string
	blank?: boolean
	icon?: string
}

const defaultProps = {
	href: '/test',
	linkType: 'external',
	disabled: false,
	size: 'md',
	className: '',
	children: undefined,
	label: '',
	blank: false,
}

export type { propsConfig }
export { defaultProps }
