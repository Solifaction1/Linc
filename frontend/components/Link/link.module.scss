@import "styles/vars";
@import "styles/utilities";

.link {
  display: flex;

  font-size: $font-size-text-md;
  font-family: $font-family-text;
  text-decoration: underline;

  &.external{
    .externalIcon{
      display: block;
      fill: $color-gray;
    }

    &:active{    
      .externalIcon{
        fill: $color-beige-dark;    
      }
    }

    &.sm{
      font-size: $font-size-text-sm;

      .externalIcon{
        max-width:  $font-size-text-sm;
        max-height: $font-size-text-sm;
      }
    }
    
    &.md{
      font-size: $font-size-text-md;

      .externalIcon{
        max-width: $font-size-text-md;
        max-height: $font-size-text-md;
      }
    }
    
    &.lg{
      font-size: $font-size-text-lg;
      
      .externalIcon{
        max-width: $font-size-text-lg;
        max-height: $font-size-text-lg;
      }
    }
  }

  &.disabled{
    opacity: .5;
    pointer-events: none;
  }

  &.navigation{
    color: $color-gray-shade-50;

    font-size: $font-size-title-md;
    font-weight: $font-weight-bold;
    text-decoration: none;
  }
}
