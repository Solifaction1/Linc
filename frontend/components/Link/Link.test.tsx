import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Link from './Link'

describe('Link', () => {
	it('renders an external Link', () => {
		const { container } = render(
			<Link href="href" disabled size="lg" className="className" label="label">
				children
			</Link>
		)

		expect(container).toMatchSnapshot()

		const link = container.getElementsByClassName(
			'link className external lg disabled'
		)[0]
		expect(link).toBeVisible()
		expect(link?.textContent).toEqual('childrenlabel')
		expect(link.getAttribute('href')).toEqual('href')
		expect(link.getAttribute('rel')).toEqual('noreferrer')
		expect(link.getAttribute('target')).toEqual('_self')
	})

	it('renders a navigation Link', () => {
		const { container } = render(
			<Link
				href="href"
				linkType="navigation"
				className="className"
				blank
				label="Hello"
			>
				{/* TODO: blank lijkt geen svg te renderen  */}
				children
			</Link>
		)

		expect(container).toMatchSnapshot()

		const link = container.getElementsByClassName(
			'link className navigation md false'
		)[0]
		expect(link).toBeVisible()
		expect(link?.textContent).toEqual('childrenHello')
		expect(link.getAttribute('href')).toEqual('href')
	})
})
