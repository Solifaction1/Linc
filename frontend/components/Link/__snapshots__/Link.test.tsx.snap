// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders a navigation Link 1`] = `
<div>
  <a
    class="link className navigation md false"
    href="href"
  >
    children
    Hello
  </a>
</div>
`;

exports[`Link renders an external Link 1`] = `
<div>
  <a
    class="link className external lg disabled"
    href="href"
    rel="noreferrer"
    target="_self"
  >
    children
    label
  </a>
</div>
`;
